pkg/migrate/migrate.go

```diff
@@ -134,6 +192,82 @@ func Migrate(ctx context.Context, w *log.Logger, clientset k8sclient.Interface,
 	return nil
 }
 
+// NewPVMigrator returns a PV migration context
+func NewPVMigrator(cfg *rest.Config, log *log.Logger, srcSC, dstSC string, podReadyTimeout int) (*PVMigrator, error) {
+	k8scli, err := k8sclient.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kubernetes client: %w", err)
+	}
+
+	if srcSC == "" {
+		return nil, fmt.Errorf("empty source storage class")
+	}
+	if dstSC == "" {
+		return nil, fmt.Errorf("empty destination storage class")
+	}
+	if log == nil {
+		return nil, fmt.Errorf("no logger provided")
+	}
+
+	podReadinessTimeout := time.Duration(podReadyTimeout)
+	if podReadinessTimeout == 0 {
+		// default
+		podReadinessTimeout = 60 * time.Second
+	}
+
+	return &PVMigrator{
+		ctx:             context.Background(),
+		log:             log,
+		k8scli:          k8scli,
+		srcSc:           srcSC,
+		dstSc:           dstSC,
+		deletePVTimeout: 5 * time.Minute,
+		podReadyTimeout: podReadinessTimeout * time.Second,
+	}, nil
+}
+
+// PrintPVAccessModeErrors prints and formats the volume access mode errors in pvcErrors
+func PrintPVAccessModeErrors(pvcErrors map[string]map[string]PVCError) {
```

Comment:
should this be private?

Reply:
I made it public for clients to have an option to conveniently print the errors in a table format.

Reply:
Why not have a unified function that prints the results to a provided output stream if it's nonnil?